### PR TITLE
Charlie/passkey registration

### PR DESCRIPTION
**Pull request by charliescheer**
### Fix
We are adding passkey support to SNMac!  This will come in a couple of stages, first in this PR we are adding the ability to register a passkey on your account.  Signing in with a passkey will come later.

With a signed in account, from the preferences view there is a new button "Add passkey sign in" tapping that will allow a user to authorize there account and create a new passkey in their iCloud Keychain.  Once added they should be able to use this passkey to sign into simplenote on Mac or iOS with the same passkey as long as that version of simplenote is updated with the passkey auth info.

### Test
1. Sign in to a simplenote account and open the preferences view
2. tap on the Add Passkey Sign in
3. Confirm that a blocking spinner appears while we fetch a challenge
4. If successful you should see an apple auth appear.  Dismiss it without verifying.  Confirm you see a failure alert and the spinner disappears
5. Tapp on the add passkey button again and this time when the apple auth appears continue to add a passkey.  When it is complete you should see a success message.  
6. Go to the passkey db and confirm your new passkey was added (ping me for a link if need be)

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:

> These changes do not require release notes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Simplenote/BlockingActivityIndicator.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    private func setupView() {
+        wantsLayer = true
+        layer?.backgroundColor = NSColor.black.withAlphaComponent(0.3).cgColor
+        translatesAutoresizingMaskIntoConstraints = false
```
Comment (jleandroperez) [inline Simplenote/BlockingActivityIndicator.swift]: Nitpicky: This one should probably live in `showIndicator`.

Usually tAMIC goes in the spot where the view is attached.

File: Simplenote/PreferencesViewController.swift [hunk 2]
```diff
@@ -320,6 +329,63 @@ class PreferencesViewController: NSViewController {
...
+    private func presentPasskeyRegistrationAlert() {
+        let alert = NSAlert(messageText: PasskeyAuthentication.alertTitle, informativeText: PasskeyAuthentication.message)
+        let passwordField = NSSecureTextField(frame: NSRect(x: 0, y: 0, width: 200, height: 24))
+        alert.accessoryView = passwordField
+        alert.addButton(withTitle: PasskeyAuthentication.submit)
+        alert.addButton(withTitle: PasskeyAuthentication.cancel)
+
+        let modalResult = alert.runModal()
+
+        if modalResult == .alertFirstButtonReturn {
```
Comment (jleandroperez) [inline Simplenote/PreferencesViewController.swift]: Nit: The alert instantiation itself could live in `NSAlert+Passkey`

**Review by jleandroperez — APPROVED** [review]
Q: I'm not sure if this is not supported on the mac, but I've been unable to store the newly registered Passkey in 1Password (I **had to** use iCloud Keychain). Is that expected?

Regarding the flow, a spinner in the Preferences UI while passkey registration goes on feels off. 

We should probably reach out to Design, it feels like the entire Passkey Registration should happen on its own window (or modal view).

Other than that, :shipit: !!

**Comment by charliescheer** [comment]
```
I'm not sure if this is not supported on the mac, but I've been unable to store the newly registered Passkey in 1Password (I had to use iCloud Keychain). Is that expected?
```

I had not looked into this because my 1password is old, apparently.  It looks like right now this is not supported in Mac, maybe next version?  I will add to my to do list to take a look at 1password support for iOS and Mac as part of the passkey project

**Pull request merged**